source/types.ts

```diff
@@ -153,6 +167,20 @@ export type Store = {
 	 */
 	resetKey: (key: string) => Promise<void> | void
 
+	/**
+	 * Method to fetch a client's hit count and reset time.
+	 *
+	 * @param key {string} - The identifier for a client.
+	 *
+	 * @returns {ClientRateLimitInfo} - The number of hits and reset time for that client.
+	 */
+	fetchKey: (
```

Comment:
I think it should be optional so that older stores aren't invalidated by this change.

Also, I feel like `get()` might be a better name; I feel like "fetch" implies a network which isn't necessarily happening here, and then dropping 'key' makes it more inline with `increment()` and `decrement()`. 

(I know we have `resetKey()`, but that is to differentiate it from `resetAll()`. Otherwise I might have just called it `reset()`)

```suggestion
	get?: (
```

Reply:
> I think it should be optional so that older stores aren't invalidated by this change.

Yea I was thinking the same 😅 

> Also, I think get() might be a better name; I feel like "fetch" implies a network request, which isn't necessarily happening here, and then dropping 'key' makes it more inline with increment() and decrement().

Okay!